### PR TITLE
v6.0.0-alpha.4

### PR DESCRIPTION
**Pull request by alexfauquette**
### Prepare the release of the packages

The following steps must be proposed as a pull request.

- [x] Compare the last tag with the branch upon which you want to release (`next` for the alpha / beta releases and `master` for the current stable version).
      To do so, use `yarn release:changelog` The options are the following:

  ```bash
  yarn release:changelog
     --githubToken   YOUR_GITHUB_TOKEN (needs "public_repo" permission)
     --lastRelease   The release to compare against (default: the last one)
     --release       The branch to release (default: master)
  ```

  You can also provide the github token by setting `process.env.GITHUB_TOKEN` variable.

  In case of a problem, another method to generate the changelog is available at the end of this page.

- [x] Clean the generated changelog, to match the format of [https://github.com/mui/mui-x/releases](https://github.com/mui/mui-x/releases).
- [x] Update the root `package.json`'s version
- [x] Update the versions of the other `package.json` files and of the dependencies with `yarn release:version`.
- [x] Fix manually the package version in `x-date-pickers/package.json` and `x-date-pickers-pro/package.json`.
- [x] Open PR with changes and wait for review and green CI.
- [ ] Merge PR once CI is green, and it has been approved.

### Release the packages

- [ ] Checkout the last version of the working branch
- [ ] Make sure you have the latest dependencies installed: `yarn`.
- [ ] Build the packages: `yarn release:build`.
- [ ] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [ ] Push the newly created tag: `yarn release:tag`.

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [ ] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```sh
git push upstream next:docs-next -f
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-v5)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-v5` deployment.

### Publish GitHub release

- [ ] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,137 @@
...
+- [DataGrid] Fix grid not updating state on `rowCount` prop change (#5982) @cherniavskii
+- [DataGrid] Rename selection props (#6556) @m4theushw
+- [l10n] Improve Turkish (tr-TR) locale on the data grid and pickers (#6542) @ramazansancar
+
+### `@mui/x-date-pickers@v6.0.0-alpha.4` / `@mui/x-date-pickers-pro@v6.0.0-alpha.4`
+
+#### Breaking changes
+
+- The `ToolbarComponent` has been replaced by a `Toolbar` component slot.
+You can find more information about this pattern in the [MUI Base documentation](https://mui.com/base/getting-started/usage/#shared-props):
```
Comment (flaviendelangle) [inline CHANGELOG.md:47]:
```suggestion
  You can find more information about this pattern in the [MUI Base documentation](https://mui.com/base/getting-started/usage/#shared-props):
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,137 @@
...
+
+- The `onChange` / `openView` props on the toolbar has been renamed `onViewChange` / `view`
+
+#### Changes
+
+- [fields] Add a `validationError` property to the `onChange` callback (#6539) @flaviendelangle
+- [fields] Distinguish start and end input error on multi input fields (#6503) @flaviendelangle
+- [pickers] Clean the `Tabs` component slot (#6543) @flaviendelangle
+- [pickers] Fix localisation of the placeholder (#6547) @alexfauquette
+- [pickers] Fix TS type issues (#6322) @flaviendelangle
```
Comment (flaviendelangle) [inline CHANGELOG.md:112]:
```suggestion
- [pickers] Fix typescript issues (#6322) @flaviendelangle
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,137 @@
...
+
+  ```diff
+  // Same on all other pickers
+  <DatePicker
+  -  ToolbarComponent: MyToolbar,
+  +  components={{ Toolbar: MyToolbar }}
+  />
+  ```
+
+- The `toolbarPlaceholder` and `toolbarFormat` props have moved to the `toolbar` components props slot:
```
Comment (LukasTy) [inline CHANGELOG.md:57]:
```suggestion
- The `toolbarPlaceholder` and `toolbarFormat` props have been moved to the `toolbar` components props slot:
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,137 @@
...
+  +  componentsProps={{
+  +    toolbar: {
+  +      toolbarPlaceholder: "__",
+  +      toolbarFormat: "DD / MM / YYYY",
+  +    }
+  +  }}
+  />
+  ```
+
+- The `toolbarTitle` prop has moved to the localization object:
```
Comment (LukasTy) [inline CHANGELOG.md:73]:
```suggestion
- The `toolbarTitle` prop has been moved to the localization object:
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,137 @@
...
+  -    dateTimePickerDefaultToolbarTitle: 'Date Time Picker',
+  +    dateTimePickerToolbarTitle: 'Date Time Picker',
+
+  -    dateRangePickerDefaultToolbarTitle: 'Date Range Picker',
+  +    dateRangePickerToolbarTitle: 'Date Range Picker',
+    }}
+  />
+  ```
+
+- The `onChange` / `openView` props on the toolbar has been renamed `onViewChange` / `view`
```
Comment (LukasTy) [inline CHANGELOG.md:104]:
```suggestion
- The `onChange` / `openView` props on the toolbar have been renamed `onViewChange` / `view`
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,137 @@
...
+### Docs
+
+- [docs] Enable inlined preview for disabled date picker (#6477) @oliviertassinari
+- [docs] Fix 404 errors (#6541) @alexfauquette
+- [docs] Fix broken links on field pages (#6501) @flaviendelangle
+- [docs] Improve markdownlint (#6518) @oliviertassinari
+
+### Core
+
+- [core] Avoid slower CI run statues @oliviertassinari
```
Comment (LukasTy) [inline CHANGELOG.md:126]:
```suggestion
- [core] Run CodeQL only on schedule @oliviertassinari
```

File: benchmark/package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "benchmark",
-  "version": "6.0.0-alpha.0",
+  "version": "6.0.0",
```
Comment (LukasTy) [inline benchmark/package.json:3]: Mistake or you want to avoid updating this non-releasing package every release? 🙈

Comment (alexfauquette) [inline benchmark/package.json:3]:
```suggestion
  "version": "6.0.0-alpha.0",
```

**Review by LukasTy — APPROVED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,137 @@
...
+- [docs] Enable inlined preview for disabled date picker (#6477) @oliviertassinari
+- [docs] Fix 404 errors (#6541) @alexfauquette
+- [docs] Fix broken links on field pages (#6501) @flaviendelangle
+- [docs] Improve markdownlint (#6518) @oliviertassinari
+
+### Core
+
+- [core] Run CodeQL only on schedule @oliviertassinari
+- [core] Fix trailing spaces and git diff format (#6523) @oliviertassinari
+- [core] Harden GitHub Actions permissions (#6396) @step-security-bot
```
[the remark this answers is not in the document]
Comment (alexfauquette) [reply CHANGELOG.md:128]: I kept it because it's not just a version bump, it modifies something that we could want to find

**Review by flaviendelangle — APPROVED** [review]

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,137 @@
...
+  | `disableMultipleSelection` | `disableMultipleRowSelection` |
+
+- The `gridSelectionStateSelector` selector was renamed to `gridRowSelectionStateSelector`.
+
+- The `selectionChange` event was renamed to `rowSelectionChange`.
+
+#### Changes
+
+- [DataGrid] Add `searchPredicate` prop to `GridColumnsPanel` component (#6557) @cherniavskii
+- [DataGrid] Allows keyboard navigation in group header (#5947) @alexfauquette
```
Comment (cherniavskii) [inline CHANGELOG.md:37]:
```suggestion
- [DataGrid] Support keyboard navigation in column group header (#5947) @alexfauquette
```

**Review by cherniavskii — APPROVED** [review]

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,137 @@
...
+  ```
+
+- The `onChange` / `openView` props on the toolbar have been renamed `onViewChange` / `view`
+
+#### Changes
+
+- [fields] Add a `validationError` property to the `onChange` callback (#6539) @flaviendelangle
+- [fields] Distinguish start and end input error on multi input fields (#6503) @flaviendelangle
+- [pickers] Clean the `Tabs` component slot (#6543) @flaviendelangle
+- [pickers] Fix localisation of the placeholder (#6547) @alexfauquette
```
[the remark this answers is not in the document]
Comment (alexfauquette) [reply CHANGELOG.md:111]: Maybe I should add vale on the changelog file too 😸

**Pull request merged**